modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx

```diff
@@ -216,139 +218,80 @@ export default function EditObjectField({
-					const dateJson: ObjectFieldDateRangeFilterSettings = {};
-
-					valueList?.forEach(({label, value}) => {
-						dateJson[value] = label;
-					});
+			else if (
+				objectFieldBusinessType === 'Picklist' ||
+				objectFieldName === 'status'
+			) {
+				let picklistJson:
+					| ExcludesFilterOperator
+					| IncludesFilterOperator;
 
-					newFilterValues = [
-						...(filter.value as ObjectFieldFilterSetting[]),
-						{
-							filterBy: objectFieldName,
-							filterType,
-							json: dateJson,
-						},
-					];
-				}
-				else if (
-					objectFieldBusinessType === 'Picklist' ||
-					objectFieldName === 'status'
-				) {
-					let picklistJson:
-						| ExcludesFilterOperator
-						| IncludesFilterOperator;
-
-					if (filterType === 'excludes') {
-						picklistJson = {
-							not: {
-								in: valueList?.map(({value}) => value) as
-									| string[]
-									| number[],
-							},
-						};
-					}
-					else {
-						picklistJson = {
+				if (filterType === 'excludes') {
+					picklistJson = {
+						not: {
 							in: valueList?.map(({value}) => value) as
 								| string[]
 								| number[],
-						};
-					}
-
-					newFilterValues = [
-						...(filter.value as ObjectFieldFilterSetting[]),
-						{
-							filterBy: objectFieldName,
-							filterType,
-							json: picklistJson,
 						},
-					];
+					};
 				}
 				else {
-					newFilterValues = [
-						...(filter.value as ObjectFieldFilterSetting[]),
-						{
-							filterBy: objectFieldName,
-							filterType,
-							json: {
-								[filterType as string]: value
-									? value
-									: valueList?.map(({value}) => value),
-							},
-						},
-					];
+					picklistJson = {
+						in: valueList?.map(({value}) => value) as
+							| string[]
+							| number[],
+					};
 				}
 
-				const newFilter: ObjectFieldSetting = {
-					name: filter.name,
-					value: newFilterValues,
-				};
-
-				const newObjectFieldSettings:
-					| ObjectFieldSetting[]
-					| undefined = [
-					...(objectFieldSettings?.filter(
-						(fieldSetting) => fieldSetting.name !== 'filters'
-					) as ObjectFieldSetting[]),
-					newFilter,
+				newFilterValues = [
+					...(filter.value as ObjectFieldFilterSetting[]),
+					{
+						filterBy: objectFieldName,
+						filterType,
+						json: picklistJson,
+					},
 				];
-
-				setAggregationFilters(newAggregationFilters);
-				setValues({
-					objectFieldSettings: newObjectFieldSettings,
-				});
 			}
+			else {
+				newFilterValues = [
+					...(filter.value as ObjectFieldFilterSetting[]),
+					{
+						filterBy: objectFieldName,
+						filterType,
+						json: {
+							[filterType as string]: value
+								? value
+								: valueList?.map(({value}) => value),
+						},
+					},
+				];
+			}
+
+			const newFilter: ObjectFieldSetting = {
+				name: filter.name,
+				value: newFilterValues,
+			};
+
+			const newObjectFieldSettings: ObjectFieldSetting[] | undefined = [
+				...(objectFieldSettings?.filter(
```

Comment:
Hi @danvb112. If I am not mistaken, this will never be `undefined`. Am I right? If so, we could remove the `| undefined` in the next PRs, since this PR has passed on relevant already. Thanks.